WordPress/src/main/java/org/wordpress/android/ui/sitemonitor/SiteMonitorParentViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package org.wordpress.android.ui.sitemonitor
+
+import androidx.compose.runtime.MutableState
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+@HiltViewModel
+class SiteMonitorParentViewModel @Inject constructor(
+    @param:Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    private val metricsViewModel: SiteMonitorTabViewModelSlice,
+    private val phpLogViewModel: SiteMonitorTabViewModelSlice,
+    private val webServerViewModel: SiteMonitorTabViewModelSlice
+) : ScopedViewModel(bgDispatcher) {
+    private lateinit var site: SiteModel
+
+    init {
+        metricsViewModel.initialize(viewModelScope)
+        phpLogViewModel.initialize(viewModelScope)
+        webServerViewModel.initialize(viewModelScope)
+    }
+
+    fun start(site: SiteModel) {
+        this.site = site
+        trackActivityLaunched()
+        metricsViewModel.start(SiteMonitorType.METRICS, SiteMonitorTabItem.Metrics.urlTemplate, site)
+        phpLogViewModel.start(SiteMonitorType.PHP_LOGS, SiteMonitorTabItem.PHPLogs.urlTemplate, site)
+        webServerViewModel.start(SiteMonitorType.WEB_SERVER_LOGS, SiteMonitorTabItem.WebServerLogs.urlTemplate, site)
+    }
+
+    fun getUiState(siteMonitorType: SiteMonitorType): MutableState<SiteMonitorUiState> {
```

Comment:
Would it be better to return here State instead of MutableState? I guess the consume is not going to modify the received state. WDYT @AjeshRPai ?

Reply:
Nice catch @pantstamp, You are right. Its better to return state here. Done in [f6633cf](https://github.com/wordpress-mobile/WordPress-Android/pull/20077/commits/f6633cf9d849a39afd0328dadcb88138504cae3b)